Classes/Private/Manager/Objects/OCTFileInput.m

```diff
@@ -0,0 +1,161 @@
+//
+//  OCTFileInput.m
+//  objcTox
+//
+//  Created by stal on 5/8/2015.
+//  Copyright © 2015 Zodiac Labs. All rights reserved.
+//
+
+#include <sys/stat.h>
+#import "DDLog.h"
+#import "OCTSubmanagerFiles.h"
+
+#undef LOG_LEVEL_DEF
+#define LOG_LEVEL_DEF LOG_LEVEL_DEBUG
+
+@interface OCTFileInput ()
+
+@property (copy)   NSString *path;
+@property (strong) NSFileHandle *readHandle;
+@property NSTimeInterval modifiedTime;
+@property OCTToxFileSize knownFileSize;
+
+@end
+
+@implementation OCTFileInput
+
+- (instancetype)initWithPath:(nonnull NSString *)path
+{
+    self = [super init];
+
+    if (self) {
+        self.path = path;
+        NSError *error = nil;
+        NSDictionary *attrs = [[NSFileManager defaultManager] attributesOfItemAtPath:[path stringByResolvingSymlinksInPath] error:&error];
+
+        if (error) {
+            DDLogError(@"unable to stat file at path: %@ because: %@", path, error);
+            return nil;
+        }
+
+        self.modifiedTime = attrs.fileModificationDate.timeIntervalSince1970;
+    }
+
+    return self;
+}
+
+#pragma mark - NSCoding
+
+- (instancetype)initWithCoder:(nonnull NSCoder *)aDecoder
+{
+    self = [super init];
+
+    if (self) {
+        self.path = [aDecoder decodeObjectForKey:@"_filePath"];
+        self.modifiedTime = [aDecoder decodeDoubleForKey:@"modifiedTime"];
```

Comment:
It would be better to make these strings static constants at the top of the file